slimdiffy/autodiff.py

```diff
@@ -711,9 +711,20 @@ def __call__(self, lambda_expr: Lambda) -> Any:
         # Process equations
         for i, eq in enumerate(lambda_expr.equations):
             self.results[i] = self.visit(eq)
+            print(f"Index {i}: {self.results[i]}")  # Debug print
 
         # Use pytree to construct result
-        def get_result(idx: int) -> Any:
+        def get_result(idx: Union[int, tuple, Tracer]) -> Any:
+            print(f"Getting result for idx: {idx}")  # Debug print
+            print(f"Available indices: {list(self.results.keys())}")  # Debug print
+            if isinstance(idx, tuple):
+                # Handle tuple indices by recursively getting each element
+                return tuple(get_result(i) for i in idx)
+            if isinstance(idx, Tracer):
+                # If we get a Tracer, return it directly
+                return idx
```

Comment:
hmm...this is a pattern that I'm seeing in this PR a lot. I think this is an anti-pattern. There should be exactly one type of thing expected by most methods and we should keep the implementations simple and reduce the number of branches possible.